icon-project/loopchain
# [LC-283] Add basic integration test

# Test steps:
To run tests, follow step below...
1. clone the repository and make virtualenv 
2. `$ make develop`
3. `$ pytest testcase/integration [-v] [options]`
> **_NOTE_**: Possible [options] are:
> - `--peer-count=[int]`: execute test with [int] peers
> - `--channel-count=[int]`: execute test with [int] channels

**_... or you can simply run tests by `$ make test`_**

# Test check list
This test tries to...
- Run loopchain launcher and check stdout lines to find any exception raised
- Check all peers and channels are ready to receive transactions 
- Send transaction and get valid tx_hash
- Check Sent transactions are successfully synced between peers (also channels)

# Test script
- `conftest.py`: contains common test fixture and helper for making loopchain config.
- `test_for_fixtures.py`: contains basic tests to be run before integration test.
- `test_integration.py`: contains tests for loopchain integration scenario. 

# Note
- Ensure that there are no loopchain-related process is running!
